feat: uniform layout and naming of hints, remarks, exceptions (#660) (#860)
zalando/restful-api-guidelines: chapters/http-headers.adoc

          $ref: '#/components/(parameters|headers)/ETag'
----

> NOTE: It is a question of taste whether headers for responses are defined in

Reviewer: ```suggestion
It is a question of taste whether headers for responses are defined in
```

  indicate that the body contains a status report resource in response to the
  requested action, which is available at provided location.

> NOTE: When using the {Content-Location} header, the {Content-Type} header

Reviewer: ```suggestion
When using the {Content-Location} header, the {Content-Type} header
```

include::../includes/prefer.yaml[]
----

> NOTE: Please, copy only the behaviors into your {Prefer} header specification

Reviewer: ```suggestion
Please, copy only the behaviors into your {Prefer} header specification
```

      $ref: 'https://opensource.zalando.com/restful-api-guidelines/models/headers-1.0.0.yaml#/Idempotency-Key'
----

TIP: The key cache is not intended as request log, and therefore should
have a limited lifetime, else it could easily exceed the data resource in
size.

> NOTE: The {Idempotency-Key} header unlike other headers in this section

Reviewer: ```suggestion
Background information: The {Idempotency-Key} header unlike other headers in this section
```

|=======================================================================

> [EXCEPTION]

Reviewer: Pls. move this exception up following the above paragraph "Below, we explicitly define the list of proprietary headers usable for all services for passing through generic context information of our fashion domain (use case 1)."

====
The only exception to this guideline are the conventional
hop-by-hop `X-RateLimit-` headers which can be used as defined in <<153>>.
====

As part of the guidelines we provide the default definition of all proprietary
headers, so you can simply reference them when defining the API endpoint. For
details see <<using-headers>>.

> TIP: This guideline does not standardize proprietary headers for our

Reviewer: ```suggestion
*Hint:* This guideline does not standardize proprietary headers for our
```

** Event listeners *must* support the metadata `flow-id` from events.

> NOTE: API-Clients *must* provide {Flow-ID} when calling a service or

Reviewer: ```suggestion
API-Clients *must* provide {Flow-ID} when calling a service or
```

** input for all API called and events published during processing
** data field written for logging and tracing

> TIP: This rule also applies to application internal interfaces and events

Reviewer: ```suggestion
This rule also applies to application internal interfaces and events
```
